### PR TITLE
Feature/vdyp polygon class set changes

### PR DESCRIPTION
**Pull request by mjunkin**

### CODE REVIEW

File: vdyp-lib/vdyp-common/src/test/java/ca/bc/gov/nrs/vdyp/application/test/TestPolygon.java [hunk 1]
```diff
@@ -16,8 +17,8 @@ public static TestPolygon build(Consumer<TestPolygon.Builder> config) {
 	}
 
 	protected TestPolygon(
-			PolygonIdentifier polygonIdentifier, Optional<Float> percentAvailable, String fiz, String becIdentifier,
-			Optional<PolygonMode> mode
+			PolygonIdentifier polygonIdentifier, Optional<Float> percentAvailable, String fiz,
+			BecDefinition becIdentifier, Optional<PolygonMode> mode
```
Comment (smithkm) [inline vdyp-lib/vdyp-common/src/test/java/ca/bc/gov/nrs/vdyp/application/test/TestPolygon.java:21]: should rename this to just `bec`

File: vdyp-lib/vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/model/BaseVdypSpecies.java [hunk 2]
```diff
@@ -12,31 +13,40 @@
...
-			PolygonIdentifier polygonIdentifier, LayerType layerType, String genus, float percentGenus, Optional<I> site
+			PolygonIdentifier polygonIdentifier, LayerType layerType, String genus, int genusIndex, float percentGenus,
+			Sp64DistributionSet sp64DistributionSet, Optional<I> site
 	) {
 		this.polygonIdentifier = polygonIdentifier;
 		this.layerType = layerType;
 		this.genus = genus;
 		this.setPercentGenus(percentGenus);
 		this.site = site;
+		this.genusIndex = genusIndex;
```
Comment (smithkm) [inline vdyp-lib/vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/model/BaseVdypSpecies.java:48]: I dislike the idea of having both the string ID and the numeric index and being able to set them separately so they might end up mismatched.

Maybe if we do need both, give the builder a copy of the control map and on build it fills in whichever of the two hasn't been provided  If both are provided either check they match, or use whichever was set most recently (Both setters set the other value to empty)

**Review by smithkm — CHANGES_REQUESTED** [review]

**Comment by smithkm** [comment]
There are also a bunch of whitespace changes and the the formatter doesn't seem to have been run.

File: vdyp-lib/vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/model/BaseVdypSpecies.java (hunk 2 above)
Comment (mjunkin) [reply vdyp-lib/vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/model/BaseVdypSpecies.java:49]: I didn't like it either - glad you prompted me to think of a better way.

**Review by smithkm — APPROVED** [review]

**Pull request merged**